kore/src/Kore/Attribute/RuleIndex.hs

```diff
@@ -0,0 +1,63 @@
+{-|
+Module      : Kore.Attribute.RuleIndex
+Description : Rule index attribute
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+Maintainer  : ???@runtimeverification.com
+
+-}
+module Kore.Attribute.RuleIndex
+    ( RuleIndex (..)
+    , ruleIndexId, ruleIndexSymbol, ruleIndexAttribute
+    ) where
+
+import Control.DeepSeq
+       ( NFData )
+import Data.Default
+import GHC.Generics
+       ( Generic )
+
+import           Kore.AST.Kore
+import           Kore.Attribute.Parser
+                 ( ParseAttributes (..) )
+import qualified Kore.Attribute.Parser as Parser
+
+{- | @RuleIndex@ represents the @RuleIndex@ attribute.
```

Comment:
I would probably just say "This attribute is used in the REPL for tagging and uniquely identifying axioms and claims".